components/vision.py

```diff
@@ -177,38 +194,43 @@ def desired_turret_rotation(self) -> Rotation2d:
         return self.relative_bearing_to_best_cluster()
 
     def turret_to_servo(self, turret: Rotation2d) -> Rotation2d:
-        return turret - (self.servo_offset - self.encoder_offset)
+        return turret - (self.servo_offsets.neutral - self.encoder_offset)
 
     @property
     def turret_rotation(self) -> Rotation2d:
         return self.raw_encoder_rotation() - self.encoder_offset
 
     def robot_to_camera(self, timestamp: float) -> Transform3d:
-        robot_to_turret_rotation = self.robot_to_turret.rotation()
         turret_rotation = self.turret_rotation_buffer.sample(timestamp)
         if turret_rotation is None:
             return self.robot_to_turret
-        return Transform3d(
-            self.robot_to_turret.translation(),
-            Rotation3d(
-                robot_to_turret_rotation.x,
-                robot_to_turret_rotation.y,
-                robot_to_turret_rotation.z + turret_rotation.radians(),
-            ),
+        r2t = self.robot_to_turret.translation()
+        t2c = self.turret_to_camera.translation().rotateBy(Rotation3d(turret_rotation))
+        trans = Translation3d(r2t.x + t2c.x, r2t.y + t2c.y, r2t.z + t2c.z)
+        rot = (
+            self.robot_to_turret.rotation()
+            .rotateBy(Rotation3d(turret_rotation))
+            .rotateBy(self.turret_to_camera.rotation())
         )
+        return Transform3d(trans, rot)
```

Comment:
I suspect much of this could be written as a sum of Transform3ds rather than individually manipulating the translations and rotations. I haven't sat down to do the maths yet though.

Reply:
I couldn't see how to do it directly on `Transform3d`s